### ISSUE TITLE
Kubelet --cloud-provider tag is not Supported on GCE with the CNI Plugin

### ISSUE THREAD

**Issue by denverwilliams** [cncf/cross-cloud#35]
When kubelet has the Cloud Provider Tag set on GCE and the CNI Plugin is being used the kube-controller-manager RouteController attempts to create routes, instead of delegating the responsibility to the CNI network plugin, making nodes unavailable for scheduling.

```
bash-4.3# kubectl describe pods kube-dns-v20-mzrgb --namespace=kube-system
Name:		kube-dns-v20-mzrgb
Namespace:	kube-system
Node:		/
Labels:		k8s-app=kube-dns
		version=v20
Status:		Pending
Tolerations:	CriticalAddonsOnly=:Exists
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath	Type		Reason			Message
  ---------	--------	-----	----			-------------	--------	------			-------
  50s		19s		7	{default-scheduler }			Warning		FailedScheduling	no nodes available to schedule pods
```

**Comment by dlx** [comment]
Looks like this is the related issue on Kubernetes https://github.com/kubernetes/kubernetes/issues/44254

**Comment by dlx** [comment]
Looks like this is a unmerged fix https://github.com/kubernetes/kubernetes/pull/33573/files
@hh might be worth integrating this into our builds, but lets not worry about it yet.

**Comment by dlx** [comment]
Have switched to using kubenet to provide networking until this is fixed so we can use --cloud-provider=gce
```
--network-plugin=kubenet \
--pod-cidr=${ pod_cidr } \
--non-masquerade-cidr=${ cidr } \
```
bash-4.3# kubectl get pods --namespace=kube-system
NAME                                                          READY     STATUS    RESTARTS   AGE
kube-apiserver-test-master1.c.test-163823.internal            1/1       Running   0          12m
kube-apiserver-test-master2.c.test-163823.internal            1/1       Running   0          12m
kube-apiserver-test-master3.c.test-163823.internal            1/1       Running   0          12m
kube-controller-manager-test-master1.c.test-163823.internal   1/1       Running   0          12m
kube-controller-manager-test-master2.c.test-163823.internal   1/1       Running   0          12m
kube-controller-manager-test-master3.c.test-163823.internal   1/1       Running   0          12m
kube-dns-v20-nksj1                                                            3/3       Running   0          11m
kube-proxy-test-master1.c.test-163823.internal                1/1       Running   0          12m
kube-proxy-test-master2.c.test-163823.internal                1/1       Running   0          12m
kube-proxy-test-master3.c.test-163823.internal                1/1       Running   0          11m
kube-scheduler-test-master1.c.test-163823.internal            1/1       Running   1          12m
kube-scheduler-test-master2.c.test-163823.internal            1/1       Running   0          11m
kube-scheduler-test-master3.c.test-163823.internal            1/1       Running   0          11m
```

**Comment by dlx** [comment]
Temporary work around for now is not to use the cloud-provider flag 

```
bash-4.3# kubectl get nodes
NAME                                  STATUS                     AGE
test-master1.c.test-163823.internal   Ready,SchedulingDisabled   15s
test-master2.c.test-163823.internal   Ready,SchedulingDisabled   13s
test-worker1.c.test-163823.internal   Ready                      16s
test-worker2.c.test-163823.internal   Ready                      16s
test-worker3.c.test-163823.internal   Ready                      10s
bash-4.3# kubectl create -f ../kube-dns.yml 
service "kube-dns" created
replicationcontroller "kube-dns-v20" created
bash-4.3# kubectl get nodes
NAME                                  STATUS                     AGE
test-master1.c.test-163823.internal   Ready,SchedulingDisabled   29s
test-master2.c.test-163823.internal   Ready,SchedulingDisabled   27s
test-master3.c.test-163823.internal   Ready,SchedulingDisabled   12s
test-worker1.c.test-163823.internal   Ready                      30s
test-worker2.c.test-163823.internal   Ready                      30s
test-worker3.c.test-163823.internal   Ready                      24s
bash-4.3# kubectl get pods --namespace=kube-system
NAME                                                          READY     STATUS    RESTARTS   AGE
kube-apiserver-test-master1.c.test-163823.internal            1/1       Running   0          40s
kube-apiserver-test-master2.c.test-163823.internal            1/1       Running   0          39s
kube-apiserver-test-master3.c.test-163823.internal            1/1       Running   0          30s
kube-controller-manager-test-master1.c.test-163823.internal   1/1       Running   0          40s
kube-dns-v20-w1b9j                                            3/3       Running   0          15s
kube-proxy-test-master2.c.test-163823.internal                1/1       Running   0          39s
kube-proxy-test-worker1.c.test-163823.internal                1/1       Running   0          39s
kube-scheduler-test-master3.c.test-163823.internal            1/1       Running   0          30s
```

**Comment by taylor** [comment]
Fixed in 127b57fee87b7799bb841f754fd32541dfefc25d
